sup3r/pipeline/forward_pass.py

```diff
@@ -746,65 +812,77 @@ def run(self):
         This routine runs forward passes on all data chunks associated with
         this file subset.
         """
-        logger.info(
-            f'Starting forward passes on data shape {self.data_shape}. Using '
-            f'{len(self.lr_slices)} chunks each with shape of '
-            f'{self.chunk_shape}, spatial_overlap of '
-            f'{self.strategy.spatial_overlap} and temporal_overlap of '
-            f'{self.strategy.temporal_overlap}')
-
-        data = np.zeros(
-            (self.strategy.s_enhance * self.data_shape[0],
-             self.strategy.s_enhance * self.data_shape[1],
-             self.strategy.t_enhance * self.data_shape[2], 2),
-            dtype=np.float32)
-
-        if self.strategy.pass_workers == 1:
-            for s_high, s_low_pad, s_high_crop in zip(self.hr_slices,
-                                                      self.lr_pad_slices,
-                                                      self.hr_crop_slices):
-
-                data_chunk = self.data_handler.data[s_low_pad]
-                data[s_high] = ForwardPass.forward_pass_chunk(
-                    data_chunk, crop_slices=s_high_crop,
-                    model_path=self.model_path)
+        if self.cache_file is not None and os.path.exists(self.cache_file):
+            logger.info('Loading forward pass output from cache file: '
+                        f'{self.cache_file}')
+            with open(self.cache_file, 'rb') as f:
```

Comment:
Caching the output in a pkl? This seems unnecessary... Isnt the output saved to h5/nc? I feel like we should assume that the forward pass always needs to be run unless the pipeline/status module has a successful flag.